src/odbc/odbc.c

```diff
@@ -1453,52 +1506,93 @@ _SQLBindParameter(SQLHSTMT hstmt, SQLUSMALLINT ipar, SQLSMALLINT fParamType, SQL
 		ODBC_EXIT_(stmt);
 	}
 
-	/* fill APD related fields */
-	apd = stmt->apd;
-	orig_apd_size = apd->header.sql_desc_count;
-	if (ipar > apd->header.sql_desc_count && desc_alloc_records(apd, ipar) != SQL_SUCCESS) {
-		odbc_errs_add(&stmt->errs, "HY001", NULL);
-		ODBC_EXIT_(stmt);
-	}
-	drec = &apd->records[ipar - 1];
+	/* If the parameter focus is set to 0, bind values as arguments */
+	/* Otherwise, bind values as the columns of a table-valued parameter */
+	if (stmt->param_focus == 0) {
```

Comment:
Wondering it it would be better to revert this test and handle directly to reduce diff change and indentation. Also could be a separate function.